lib/features/opening_hours/domain/entities/timeslot.dart

```diff
@@ -31,7 +33,17 @@ extension TimeOfDayOperators on TimeOfDay {
     }
   }
 
+  /// Checks if [this] is between [start] and [end].
+  ///
+  /// Takes into account that [start] can be after [end] (crossing midnight).
+  bool isBetween(TimeOfDay start, TimeOfDay end) {
+    return start <= end
+        ? start <= this && this <= end
+        : start <= this || this <= end;
+  }
```

Comment:
This _might_ have been a little easier to read if we weren't comparing "TimeOfDay" but an actual DateTime object, but if we **need** to use TimeOfDay  this is fine